src/NServiceBus.Core/Routing/SpecificInstanceDistributionPolicy.cs

```diff
@@ -0,0 +1,42 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Routing;
+
+    class SpecificInstanceDistributionPolicy : IDistributionPolicy
+    {
+        readonly string specificInstance;
+
+        public SpecificInstanceDistributionPolicy(string specificInstance)
+        {
+            this.specificInstance = specificInstance;
+        }
+
+        public DistributionStrategy GetDistributionStrategy(string endpointName)
+        {
+            return new SpecificInstanceDistributionStrategy(specificInstance);
+        }
+
+        class SpecificInstanceDistributionStrategy : DistributionStrategy
+        {
+            public SpecificInstanceDistributionStrategy(string specificInstance)
+            {
+                this.specificInstance = specificInstance;
+            }
+
+            public override IEnumerable<UnicastRoutingTarget> SelectDestination(IList<UnicastRoutingTarget> allInstances)
+            {
+                var target = allInstances.FirstOrDefault(t => t.Instance != null && t.Instance.Discriminator == specificInstance);
+                if (target == null)
+                {
+                    throw new Exception($"Specified instance {specificInstance} has not been configured in the routing tables.");
+                }
+                yield return target;
```

Comment:
would be interesting to measure this. Since it's a public API and users can implement their own strategies, it probably makes sense to return an enumerable instead of a List?